src/Components/WaitingRoom.tsx

```diff
@@ -38,39 +50,123 @@ const WaitingRoom = ({isHost}: {isHost: boolean}) => {
     gameDispatch({type: "startGame"})
   }
 
-  const backToLobby = () => {
-    gameDispatch({ type: "exitRoom" })
+  const style = {
+    position: 'absolute' as 'absolute',
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)',
+    width: 400,
+    bgcolor: 'background.paper',
+    border: '2px solid #000',
+    boxShadow: 24,
+    p: 4,
+  };
+
+  const exitRoom = async () => {
+    if (players.length === 1) {
+      deleteRoom()
+    } else {
+      // ! Feel like there could be async issues below but need revision
+      // * remove current player from player list 
+      const currentPlayerIndex: number = players.findIndex((player: any) => player.number === currentPlayer.number)
+      console.log("current player index",currentPlayerIndex)
+      const updatePlayers = [...players]
+      updatePlayers.splice(currentPlayerIndex, 1)
+      console.log("updated players", updatePlayers)
+      
+      // changing host number
+      if (currentPlayer.number === host) {
+        await updateDoc(doc(gameState.roomId), {
```

Comment:
this is fine, but you can also achieve the same thing by
using the helper methods in `src/utils/useFirestore.tsx`

```
await setDoc(gameState.roomId, {
      host: updatePlayers[0].number,
      players: updatePlayers
  })
```

Reply:
hmmm i tried this but it didn't work for me :'(. Did you try it too?

Reply:
🫡 yes bosssss